--- ./unix/stb_image.h
+++ ./unix/stb_image.h
@@ -329,8 +329,12 @@
 #ifdef STB_IMAGE_STATIC
 #define STBIDEF static
 #else
+#ifdef _WIN32
+#define STBIDEF extern __declspec(dllexport)
+#else
 #define STBIDEF extern
 #endif
+#endif
 
 //////////////////////////////////////////////////////////////////////////////
 //
--- ./unix/stb_image_write.h
+++ ./unix/stb_image_write.h
@@ -156,7 +156,11 @@
 #ifdef __cplusplus
 #define STBIWDEF  extern "C"
 #else
-#define STBIWDEF  extern
+#ifdef _WIN32
+#define STBIWDEF extern __declspec(dllexport)
+#else
+#define STBIWDEF extern
+#endif
 #endif
 #endif
 #endif
