--- ./scripts/mk_util.py
+++ ./scripts/mk_util.py
@@ -2597,8 +2597,8 @@
         if is_cygwin_mingw():
             # when cross-compiling with MinGW, we need to statically link its standard libraries
             # and to make it create an import library.
-            SLIBEXTRAFLAGS = '%s -static-libgcc -static-libstdc++ -Wl,--out-implib,libz3.dll.a' % SLIBEXTRAFLAGS
-            LDFLAGS = '%s -static-libgcc -static-libstdc++' % LDFLAGS
+            SLIBEXTRAFLAGS = '%s -Wl,--out-implib,libz3.dll.a' % SLIBEXTRAFLAGS
+            LDFLAGS = '%s' % LDFLAGS
         if sysname == 'Linux' and machine.startswith('armv7') or machine.startswith('armv8'):
             CXXFLAGS = '%s -fpic' % CXXFLAGS
 
