--- ./src/discover/discover.ml
+++ ./src/discover/discover.ml
@@ -3,7 +3,7 @@
 type os = Win32 | Darwin | Other
 
 let os c =
-  let win32 = C.ocaml_config_var c "os_tyle" = Some "Win32" in
+  let win32 = C.ocaml_config_var c "os_type" = Some "Win32" in
   if win32 then Win32
   else
     match C.Process.run c "uname" [ "-s" ] with
--- ./src/win32/draw.c
+++ ./src/win32/draw.c
@@ -22,8 +22,6 @@
 #include "caml/memory.h"
 
 HDC gcMetaFile;
-int grdisplay_mode;
-int grremember_mode;
 GR_WINDOW grwindow;
 
 static void GetCurrentPosition(HDC hDC,POINT *pt)
