--- ./configure
+++ ./configure
@@ -6491,7 +6491,7 @@
   CFLAGS="$CFLAGS -fPIC"
 fi
 
-OCAML_WHERE="`$OCAMLFIND ocamlc -where`"
+OCAML_WHERE="`$OCAMLFIND ocamlc -where | tr -d '\015'`"
 # Beurk!
 FINDLIB_PATH="`$OCAMLFIND printconf path | tr \\\\n \\; | sed -e 's#;$##' | sed -e 's#;#";"#g'`"
 if test -n "${FINDLIB_PATH}"; then
@@ -6556,7 +6556,7 @@
 printf "%s\n" "no" >&6; }
     fi
     OCAMLC="$OCAMLFIND ocamlopt"
-    _DEP_OPTS="-native"
+    _DEP_OPTS="-native -slash"
 else
     BYTE=yes
     OCAMLCOMP=OCAMLC
@@ -6622,7 +6622,7 @@
     else
         OCAMLC="$OCAMLFIND ocamlc"
     fi
-    _DEP_OPTS=""
+    _DEP_OPTS="-slash"
 fi
 
 OCAML_VERSION="`$OCAMLC -version`"
--- ./configure.ac
+++ ./configure.ac
@@ -217,7 +217,7 @@
   CFLAGS="$CFLAGS -fPIC"
 fi
 
-OCAML_WHERE="`$OCAMLFIND ocamlc -where`"
+OCAML_WHERE="`$OCAMLFIND ocamlc -where | tr -d '\015'`"
 # Beurk!
 FINDLIB_PATH="`$OCAMLFIND printconf path | tr \\\\n \\; | sed -e 's#;$##' | sed -e 's#;#";"#g'`"
 if test -n "${FINDLIB_PATH}"; then
@@ -275,7 +275,7 @@
         AC_MSG_RESULT(no)
     fi
     OCAMLC="$OCAMLFIND ocamlopt"
-    _DEP_OPTS="-native"
+    _DEP_OPTS="-native -slash"
 else
     BYTE=yes
     OCAMLCOMP=OCAMLC
@@ -296,7 +296,7 @@
     else
         OCAMLC="$OCAMLFIND ocamlc"
     fi
-    _DEP_OPTS=""
+    _DEP_OPTS="-slash"
 fi
 
 OCAML_VERSION="`$OCAMLC -version`"
