--- a/src/tools/main.ml
+++ b/src/tools/main.ml
@@ -103,7 +103,9 @@ let command sscmd =
       Filename.concat command_path scmd
     end in
   let scmd = "why3 " ^ sscmd in
-  Unix.execv cmd (Array.of_list (scmd :: args))
+  (* add double quotes to avoid splitting issues with execv in Windows *)
+  let quote = if Sys.win32 then Filename.quote else (fun s -> s) in
+  Unix.execv cmd (Array.of_list (List.map quote (scmd :: args)))
 
 let () = try
   let extra_help fmt () = extra_help fmt (available_commands ()) in
