--- ./emacs/merlin.el
+++ ./emacs/merlin.el
@@ -1694,7 +1694,7 @@
         ((equal merlin-command 'opam)
          (with-temp-buffer
            (if (eq (call-process-shell-command
-                    "opam config var bin" nil (current-buffer) nil) 0)
+                    (if (eq system-type 'cygwin) "cygpath \"$(opam config var bin)\"" "opam config var bin") nil (current-buffer) nil) 0)
                (let ((bin-path
                       (replace-regexp-in-string "\n$" "" (buffer-string))))
                  ;; the opam bin dir needs to be on the path, so if merlin
--- ./src/frontend/ocamlmerlin/ocamlmerlin.c
+++ ./src/frontend/ocamlmerlin/ocamlmerlin.c
@@ -539,7 +539,7 @@
     user[0] = '\0';
   // @@DRA Need to use Windows API functions to get meaningful values for st_dev and st_ino
   snprintf(eventname, PATHSZ,
-      "ocamlmerlin_%s_%lx_%llx",
+      "ocamlmerlin_%s_%lx_%I64x",
       user,
       info.dwVolumeSerialNumber,
       ((__int64)info.nFileIndexHigh) << 32 | ((__int64)info.nFileIndexLow));
